Edozwo 931

inkuss opened the pull request:
publication and de-publication of website versions
currently rolled out on edoweb-test and tested there

Commit:
Neuer Parameter crawlsPerNight anstatt Hardcoding

Commit:
Merge branch 'master' of https://github.com/edoweb/regal-api into HEAD

Commit:
make 'restricted' the standard access scheme for webpage versions

Commit:
set standard access scheme for webpages to 'restricted'

Commit:
Einhängen einer Funktion 'publishWebpageVersion'
  - sowie Unterdrückung einiger nerviger Debug-Meldungen (Stacktraces)

Commit:
save changes for rebase

Commit:
Merge branch 'master' into EDOZWO-931

Commit:
implement publishing a website version

Commit:
publish a webpage version
 - die Entwicklung ist soweit fertig. Der 1. Schritt: Anlegen einer symbolischen Verknüpfung funktioniert. Ist getestet.
   Der 2. Schritt: Ändern des Openwayback-Links von "wayback" (oder "lesesaal") auf "weltweit" scheitert noch. Der Modify (Update Resource/conf) scheitert,
   gibt eine "FileNotFoundException" auf das "uploadFile" (WARC-Archiv) aus.

Commit:
publish webpage version and update openwayback link

Commit:
refactor website version publishing
  - neue Klasse WebsiteVersionPublisher
  - Methoden aus der Klasse WebgatherUtils in die neue Klasse verschoben

Commit:
retreat webpage version

Commit:
ändere Reihenfolge Node-Update und Veröffentlichung/Retreat

Commit:
erweitere Message beim Veröffentlichen von Webschnitten

Commit:
perform publication and de-publication also on subordinate objects, if the checkmark is set

jschnasse commented on app/controllers/Resource.java:
```diff
@@ -315,11 +316,21 @@ public Result apply() {
+				return JsonMessage(new Message(result));
+			} catch (Exception e) {
+				play.Logger.error("", e);
+				return JsonMessage(new Message(json(e.toString())));
```
Vielleicht mit Error Code?
return JsonMessage(new Message(e, 500));

jschnasse commented on app/actions/Create.java:
```diff
@@ -79,8 +81,36 @@ private Node updateResource(Node node) {
+		play.Logger.debug("Patching Node with Pid " + node.getPid());
 		new Index().remove(node);
 		setNodeMembers(node, object);
+
```
Vielleicht extra Methode?

jschnasse commented on app/helper/JsonMapper.java:
```diff
@@ -532,7 +531,9 @@ private void postProcessContribution(Map<String, Object> rdf) {
 		} catch (Exception e) {
-			play.Logger.debug("Problem processing key contribution.agent", e);
+			play.Logger.debug("Problem processing key contribution.agent",
+					e.getMessage());
```
Nur die kurze Nachricht (e.getMessage()) anstatt des ganzen Stacktraces. Absicht oder nicht?

jschnasse commented on conf/application.conf.tmpl:
```diff
@@ -89,6 +89,8 @@ regal-api.webharvests.dataUrl="http://localhost/webharvests"
 #-------------wpull (Crawler)-----------------------                            
 regal-api.wpull.crawler="/usr/local/bin/wpull3"
 regal-api.wpull.jobDir="/opt/regal/wpull-data"
+regal-api.wpull.dataUrl="http://localhost"
```
Kommentar für die Installation in die Beschreibung des PR packen

jschnasse commented on app/helper/WebsiteVersionPublisher.java:
```diff
@@ -0,0 +1,428 @@
+			createSoftlinkInPublicData(node, conf);
+			setOpenwaybackLinkToPublicAccessPoint(node, conf);
+		} catch (Exception e) {
+			WebgatherLogger.error("Webpage Version " + node.getPid()
```
kommt die Error message jetzt nicht doppelt? Weil in Resource.java#patchResource auch noch einmal geloggt wird.

jschnasse commented on app/helper/WebsiteVersionPublisher.java:
```diff
@@ -0,0 +1,428 @@
+			WebgatherLogger.info(
+					"Openwayback-Link wurde auf \"weltweit\" gesetzt für Webschnitt "
+							+ node.getPid() + ". Modify-Message: " + msg);
+		} catch (UpdateNodeException e) {
```
Dieses Catch vielleicht raus?

inkuss commented:
this PR intoduces a new parameter in application.conf:
regal-api.public.jobDir="/opt/regal/public-data"

Commit:
auslagern von "handleWebpagePublishing" von Create.java nach WebsiteVersionPublisher.java
  - weitere kleinere Veränderungen gem. Pull Request #289 (siehe die Kommunikation dort)
  - auch: Parameter --no-robots in wpull-Crawling auslesen und setzen !!
